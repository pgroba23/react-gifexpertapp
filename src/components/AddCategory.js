import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ( { setCategories } ) => {

    const [inputValue, setInputValue] = useState('');

    const handleChange = ( e ) => {
        setInputValue( e.target.value );
        //console.log('Paso por aca= '+ inputValue);
    };

    const handleSubmit = ( e ) => {
        e.preventDefault();
        
        if( inputValue.trim().length > 2)
        {
            setCategories( cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }


    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
          <input 
            type="text"
            value={ inputValue }
            onChange={ handleChange }
          />  
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
