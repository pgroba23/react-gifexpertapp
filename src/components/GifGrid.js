import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

   
    return (
        <>
            <h2 className="animate__jello">{ category }</h2>
            {
                loading && <p className="animate__flash">Loading..</p>
            }
            <div className="card-grid">
                    {
                        images.map( img => 
                            <GifGridItem 
                                key= { img.id }
                                {...img} //envia el contenido como parametro
                            />
                        )
                    }
            </div>
        </>
    )
}
