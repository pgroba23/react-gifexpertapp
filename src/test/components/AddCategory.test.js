import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';


describe('Probando <AddCategory />', () => {
    
    const setCategories = () => {};

    const wrapper = shallow( <AddCategory setCategories={ setCategories }/> )

    test('debe retornar un AddCategory', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
    
});


