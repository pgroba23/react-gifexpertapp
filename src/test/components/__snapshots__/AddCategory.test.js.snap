// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <AddCategory /> debe retornar un AddCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
